process_report/process_report.py

```diff
@@ -377,6 +377,9 @@ def fetch_s3_old_pi_file():
     local_name = "PI.csv"
     invoice_bucket = get_invoice_bucket()
     invoice_bucket.download_file("PIs/PI.csv", local_name)
+    invoice_bucket.upload_file(
+        local_name, f"PIs/Archive/PI {get_iso8601_time()}.csv"
+    )  # Create backup when fetching
```

Comment:
This will try to always create this backup copy every time this program is run. We call `fetch_s3_old_pi_file` very early in the program, basically right after we parse the arguments this call is made; so even before the program does anything useful it is called.

This also ignores the `--upload-to-s3` flag because.

I think the best time to upload the backup/old pi file would be right before you create the updated version.

Reply:
I have made it such that the PI file will be backed up right before the New PI credit is applied.